### PR TITLE
Master 3.7 cf

### PR DESCRIPTION
**Pull request by mingwandroid**

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
Linux seems to be missing some kind of patch so it aint finding libs.
For windows we should copy over the compiler overrides in 3.8 to deal with the msbuild version requirement change.

**Comment by mingwandroid** [comment]
Hey @mariusvniekerk,the failure is somehow caused by enabling PGO. This is a bug that we thought was due to disabling prefix replacement but there's something else wrong too!

If I set _OPTIMIZED=no then the build works fine.

**Comment by mingwandroid** [comment]
OK I got to the bottom of the problem with embedding python and added a test for embedding with both the static and shared libpythons. Note, for the static case (but probably not for the shared case) it is essential that the exe be located in the correct relative path to the shared libpython (or the other way around really!)

Hopefully this PR will be ready soon, then I'll port the improvements over to python 3.8 and merge back to AnacondaRecipes.

Cheers.

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+python:                        # [win]
+  - 3.8                        # [win]
```
Comment (jjhelmus) [inline recipe/conda_build_config.yaml]: This should be 3.7 or removed.

Comment (mingwandroid) [reply recipe/conda_build_config.yaml]: 3.7 yeah, thanks

Comment (mariusvniekerk) [reply recipe/conda_build_config.yaml]: may as well be removed

**Comment by mariusvniekerk** [comment]
ooh windows is green

**Comment by mingwandroid** [comment]
I need to put PGO back in for Windows yet and figure out what's going on with my attempt to limit the prefix replacements on Linux. Getting there. 3.8 started too.

**Comment by mingwandroid** [comment]
Windows PGO fails:

LINK : fatal error LNK1104: cannot open file 'pgort.lib' [%SRC_DIR%\PCbuild\pythoncore.vcxproj]

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=105253&view=logs&j=21f341a4-37ca-57ed-1287-2998d263b060&t=417b2021-e743-51fa-ff2a-b94cccea1b49&l=1033

I will have to disable this again I think.

**Comment by mariusvniekerk** [comment]
@conda-forge-admin please rerender

**Comment by mingwandroid** [comment]
Drone failing here?

**Comment by mariusvniekerk** [comment]
Awesome.  Anything else you want to add before we merge?

**Comment by mingwandroid** [comment]
I think this is good to go now. There'll hopefully be more improvements (size, PGO on Windows) soon! PGO works locally for me with VS2019 FWIW.

**Comment by mariusvniekerk** [comment]
Yeah we need to just install the right additional vs2017 packages to get pgo going

**Comment by jakirkham** [comment]
Do we know what those are or where to get them?

**Comment by mariusvniekerk** [comment]
@jakirkham I've got a PowerShell snippet I'm testing out that might work

**Comment by mariusvniekerk** [comment]
Something like this seems to work https://github.com/conda-forge/_azure_pipelines_test/blob/master/azure-pipelines.yml#L90-L131
Takes about 10 minutes or so.

**Comment by mariusvniekerk** [comment]
Tried to run the install for what we thought was the required package.

See the following log for the details:
https://dev.azure.com/conda-forge/84710dde-1620-425b-80d0-4cf5baca359d/_apis/build/builds/105965/logs/56

**Comment by jakirkham** [comment]
Does that mean PGO is already working on Windows?

**Comment by mingwandroid** [comment]
No, but AFAIK we've never had PGO on windows. At least since the move to AP no python build have had PGO enabled. This isn't a regression, I'm just making people aware.

Once we figure out how to install the package, comment out set PGO= in bld.bat

If you do that before then then you'll get a build failure.

**Pull request merged**